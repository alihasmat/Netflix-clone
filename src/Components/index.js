export { default as Accordian } from "./Accordian/Accordian";
export { default as Card } from "./Card/Card";
export { default as Feature } from "./Feature/Feature";
export { default as Footer } from "./Footer/Footer";
export { default as Form } from "./Form/Form";
export { default as Header } from "./Header/Header";
export { default as Jumbotron } from "./Jumbotron/Jumbotron";
export { default as Loading } from "./Loading/Loading";
export { default as Player } from "./Player/Player";
export { default as Profiles } from "./Profiles/Profiles";
export { default as Search } from "./Search/Search";
