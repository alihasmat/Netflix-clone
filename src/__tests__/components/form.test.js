import { render, fireEvent } from "@testing-library/react";
import { Form } from "../../Components";

jest.mock("react-router-dom");

describe("<Form />", () => {
  it(`render <Form /> from populated data`, () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>Sign Up Now</Form.Title>

        <Form.Base>
          <Form.Input
            type="text"
            placeholder="First Name"
            onChange={() => {}}
          />
          <Form.Input
            type="email"
            placeholder="Email Address"
            onChange={() => {}}
          />
          <Form.Input
            type="password"
            placeholder="password"
            onChange={() => {}}
          />
          <Form.Submit disabled type="submit">
            Sign Up
          </Form.Submit>
          <Form.Text>
            Already have an account. <Form.Link to="/signin">Sign In</Form.Link>
          </Form.Text>
          <Form.SmallText>
            This page is protected by Google reCAPTCHA.
          </Form.SmallText>
        </Form.Base>
      </Form>
    );
    expect(getByText("Sign Up Now")).toBeTruthy();
    expect(getByText("Already have an account.")).toBeTruthy();
    expect(
      getByText("This page is protected by Google reCAPTCHA.")
    ).toBeTruthy();
    expect(getByPlaceholderText("First Name")).toBeTruthy();
    expect(getByPlaceholderText("Email Address")).toBeTruthy();
    expect(getByPlaceholderText("password")).toBeTruthy();
    expect(getByText("Sign Up").disabled).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it(`render <Form /> with error`, () => {
    const { container, getByText, queryByText } = render(
      <Form>
        <Form.Error>Your email address is already being used</Form.Error>
        <Form.Submit type="submit">Sign In</Form.Submit>
      </Form>
    );
    expect(getByText("Your email address is already being used")).toBeTruthy();
    expect(queryByText("Sign In").disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
