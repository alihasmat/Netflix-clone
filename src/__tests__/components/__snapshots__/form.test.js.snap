// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> render <Form /> from populated data 1`] = `
<section
  class="style__Container-sc-7c6rek-0 kWiYtN"
>
  <h1
    class="style__Title-sc-7c6rek-1 gnxZoL"
  >
    Sign Up Now
  </h1>
  <form
    class="style__Base-sc-7c6rek-2 fKwXPb"
  >
    <input
      class="style__Input-sc-7c6rek-3 FKGAe"
      placeholder="First Name"
      type="text"
    />
    <input
      class="style__Input-sc-7c6rek-3 FKGAe"
      placeholder="Email Address"
      type="email"
    />
    <input
      class="style__Input-sc-7c6rek-3 FKGAe"
      placeholder="password"
      type="password"
    />
    <button
      class="style__Submit-sc-7c6rek-4 euslZH"
      disabled=""
      type="submit"
    >
      Sign Up
    </button>
    <p
      class="style__Text-sc-7c6rek-5 hqVNiQ"
    >
      Already have an account. 
    </p>
    <p
      class="style__SmallText-sc-7c6rek-6 gjTpEe"
    >
      This page is protected by Google reCAPTCHA.
    </p>
  </form>
</section>
`;

exports[`<Form /> render <Form /> with error 1`] = `
<section
  class="style__Container-sc-7c6rek-0 kWiYtN"
>
  <div
    class="style__Error-sc-7c6rek-7 iMFMFP"
  >
    Your email address is already being used
  </div>
  <button
    class="style__Submit-sc-7c6rek-4 euslZH"
    type="submit"
  >
    Sign In
  </button>
</section>
`;
